litellm/utils.py

```diff
@@ -671,8 +672,79 @@ def post_call_processing(original_response, model, optional_params: Optional[dic
         except Exception as e:
             raise e
 
+    async def _get_mock_healthy_deployments(model):
+        """
+        Returns a mock healthy LiteLLM Router deployment for consumption during retry logic.
+
+        Returns:
+            - A list of healthy deployments containing one mock deployment corresponding to
+              the model
+            - A list of deployments containing one mock deployment corresponding to the model.
+        """
+        mock_deployment = {
+            "model_name": model,
+        }
+        return [mock_deployment], [mock_deployment]
+
+    def _get_model_from_wrapper_args(
+        call_type, empty_model_call_types, args, kwargs
+    ) -> Optional[str]:
+        """
+        Fetches the name of the model from the arguments passed to the wrapper function.
+
+        Args:
+            call_type: The name of the LiteLLM call type (text completion, image generation, etc.).
+            empty_model_call_types: A list of call types that do not require a model. If the model
+                                    is not found in the arguments, the function will return None if
+                                    the call type is in this list, else it will raise an error.
+            args: The positional arguments passed to the wrapper function.
+            kwargs: The keyword arguments passed to the wrapper function.
+        """
+        model = None
+        try:
+            model = args[0] if len(args) > 0 else kwargs["model"]
+        except Exception:
+            model = None
+            if call_type not in empty_model_call_types:
+                raise ValueError("model param not passed in.")
+        return model
+
     @wraps(original_function)
-    def wrapper(*args, **kwargs):  # noqa: PLR0915
```

Comment:
Split wrapper into an internal `_wrapper` function (with preexisting retry logic removed) and an outer function that calls  the new `run_with_retries` (synchronous) that reuses retry logic from Router